addons/stock/models/stock_move.py

```diff
@@ -416,44 +418,75 @@ def write(self, vals):
                 receipt_moves_to_reassign |= move_to_unreserve.filtered(lambda m: m.location_id.usage == 'supplier')
                 receipt_moves_to_reassign |= (self - move_to_unreserve).filtered(lambda m: m.location_id.usage == 'supplier' and m.state in ('partially_available', 'assigned'))
 
-        # TDE CLEANME: it is a gros bordel + tracking
-        Picking = self.env['stock.picking']
-
-        #propagation of expected date:
+        # Handle the propagation of `date_expected` and `date` fields.
         propagated_date_field = False
         if vals.get('date_expected'):
-            #propagate any manual change of the expected date
             propagated_date_field = 'date_expected'
-        elif (vals.get('state', '') == 'done' and vals.get('date')):
-            #propagate also any delta observed when setting the move as done
+        elif vals.get('state', '') == 'done' and vals.get('date'):
             propagated_date_field = 'date'
-
         if propagated_date_field:
-            #any propagation is (maybe) needed
+            new_date = vals.get(propagated_date_field)
             for move in self:
-                if move.move_dest_ids and move.propagate_date:
-                    new_date = vals.get(propagated_date_field)
-                    delta_days = (new_date - move.date_expected).total_seconds() / 86400
-                    if abs(delta_days) < move.propagate_date_minimum_delta:
-                        continue
-                    for move_dest in move.move_dest_ids:
-                        if move_dest.state not in ('done', 'cancel'):
-                            move_dest.date_expected += relativedelta.relativedelta(days=delta_days)
-        track_pickings = not self._context.get('mail_notrack') and any(field in vals for field in ['state', 'picking_id', 'partially_available'])
+                move_dest_ids = move.move_dest_ids.filtered(lambda m: m.state not in ('done', 'cancel'))
+                if not move_dest_ids:
+                    continue
+                delta_days = (new_date - move.date_expected).total_seconds() / 86400
+                if not move.propagate_date or abs(delta_days) < move.propagate_date_minimum_delta:
+                    move_dest_ids._delay_alert_log_activity('manual', move)
+                    continue
+                for move_dest in move_dest_ids:
+                    move_dest.date_expected += relativedelta.relativedelta(days=delta_days)
```

Comment:
You modified the behavior. It will propagate even if propagate_date is set to False